documentation/development-docs/azure-powershell-developer-guide.md

```diff
@@ -0,0 +1,451 @@
+- `AZURE_SERVICE_PRINCIPAL` should be a service principal - an application defined in the subscription's tenant - that has management access to the subscription (or at least to a resource group in the tenant)
+    - `AZURE_SERVICE_PRINCIPAL=UserId=<UserGuid>;Password=<Password>;AADTenant=<TenantGuid>;SubscriptionId=<SubscriptionId>`
+
+### Recording/Running Tests 
+
+- Set up environment variables using New-TestCredential as described [here](https://github.com/Azure/azure-powershell/blob/preview/documentation/Using-Azure-TestFramework.md)
+- [Run the test](#running-tests) and make sure you got a generated JSON file that matches the test name in the bin folder under the `SessionRecords` folder
+- Copy this `SessionRecords` folder and place it inside the test project
+    - Inside Visual Studio, add all of the generated JSON files, making sure to change the "Copy to Output Directory" property for each one to "Copy if newer"
+    -  Make sure that all of these JSON files appear in your `Commands.SERVICE.Test.csproj` file
+
+# After Development
+
+Once all of your cmdlets have been created and the appropriate tests have been added, you can open a pull request in the Azure PowerShell repository to have your cmdlets added to the next release. Please make sure to read [CONTRIBUTING.md](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md) for more information on how to open a pull request, clean up commits, make sure appropriate files have been added/changed, and more.
+
+# Misc
+
+## Publish to PowerShell Gallery
+
+- To publish your module to the [official PowerShell gallery](http://www.powershellgallery.com/), or the test gallery site, contact the Azure PowerShell team
+- To create a signed module package for local usage, use the [powershell-sign](http://azuresdkci.cloudapp.net/view/1-AzurePowerShell/job/powershell-sign/) job on Jenkins
+
+## AsJob Parameter
+
+All long running operations must implement the `-AsJob` parameter, which will allow the user to create jobs in the background. For more information about PowerShell jobs and the -AsJob parameter, read [this doc](https://docs.microsoft.com/en-us/powershell/azure/using-psjobs).
+
+To implement the `-AsJob` parameter, simply add the parameter to the end of the parameter list:
+
+````cs
+[Parameter(Mandatory = false, HelpMessage = "Run cmdlet in the background")]
+public SwitchParameter AsJob { get; set; }
+````
+
+Once you add the parameter, please manually test that the job is created and successfully completes when the parameter is specified.  Additionally, please ensure that the help files are updated with this parameter.
+
+To ensure that `-AsJob` is not broken in future changes, please add a test for this parameter. To update tests to include this parameter, use the following pattern:
+
+````powershell
+$job = Get-AzureRmSubscription
+$job | Wait-Job
+$subcriptions = $job | Receive-Job
+````
+
+## Argument Completers
```

Comment:
This feels unnecessary here

Reply:
@maddieclayton I plan on moving this information to the new document outlining PowerShell attributes and properties